### PR TITLE
Feature/import dags in nested folders

### PR DESCRIPTION
**Pull request by quentin-bayart**
This PR allows dag file processors to import dags in subfolders within a zip file. 

The dags in Intelligence Studio are defined in a subfolder and use relative imports.

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -256,6 +256,13 @@
       type: integer
       example: ~
       default: "2"
+    - name: dagbag_import_tree_depth
+      description: |
+        If dags are defined in a subfolder, how deep will a DagFileProcessor look
```
Comment (choyrim) [inline airflow/config_templates/config.yml:261]: perhaps mention that zero, the default, means look in root folder. which i assume makes it backward compatible. i assume this description ends up in the documentation.

**Review by choyrim — APPROVED** [review]
I've checked the PR for IP and confirmed that the change has none.

**Comment by choyrim** [comment]
@quentin-bayart we should also sync the fork and rebase your branch to make sure everything is ok

**Comment by quentin-bayart** [comment]
Will create a different PR with the same changes squashed in a clean one commit.

**Pull request closed without merging**